src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Alert, NativeSyntheticEvent} from 'react-native';
+import {ContextMenuOnPressNativeEvent} from 'react-native-context-menu-view';
+import {useCallback, useContext} from 'react';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import UseMessageHook from '../../../types/hooks/screens/useMessageHook.types';
+import useChatScreenHook from './useChatScreenHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {ANONYMOUS} from '../core/constant';
+import {Context} from '../../context';
+import {calculateTime} from '../../utils/time';
+import {setReplyTarget} from '../../context/actions/chat';
+
+function useMessageHook(): UseMessageHook {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {handleUserName} = useChatScreenHook(ANONYMOUS);
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+
+  const setReplyPreview = useCallback((messageItem) => {
+    console.log('Reply Message');
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    console.log('clear Reply target');
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = async (messageId: string, type: 'ANONYMOUS' | 'SIGNED') => {
+    if (!localDb) return;
+    try {
+      if (type === 'ANONYMOUS') {
+        await AnonymousMessageRepo.deleteMessage(messageId);
+      } else {
+        await SignedMessageRepo.deleteMessage(messageId);
+      }
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+    } catch (error) {
+      console.log('error on delete message:', error);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function deletes a message based on its type. It might be more maintainable to have separate functions for deleting anonymous and signed messages. This way, if the logic for deleting either type of message changes in the future, you only need to update one specific function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->